src/Microsoft.AspNetCore.Blazor.Browser.JS/src/Services/UriHelper.ts

```diff
@@ -53,11 +54,13 @@ function performInternalNavigation(absoluteInternalHref: string) {
 }
 
 async function handleInternalNavigation() {
-  await DotNet.invokeMethodAsync(
-    'Microsoft.AspNetCore.Blazor.Browser',
-    'NotifyLocationChanged',
-    location.href
-  );
+  if (notifyLocationChangedCallback) {
+    await DotNet.invokeMethodAsync(
+      notifyLocationChangedCallback!.assemblyName,
+      notifyLocationChangedCallback!.functionName,
```

Comment:
You don't need the `!` modifiers on lines 59 and 60, because the compiler will have narrowed the type automatically through flow analysis.

Inside the `if (notifyLocationChangedCallback) { }` block, the type of `notifyLocationChangedCallback` will not have the `| null` part, since the block wouldn't be entered if it was `null`.